// csp-header/dist/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
// General values
exports.NONE = "'none'";
exports.SELF = "'self'";
exports.INLINE = "'unsafe-inline'";
exports.EVAL = "'unsafe-eval'";
exports.DATA = 'data:';
exports.BLOB = 'blob:';
// script-src values
exports.STRICT_DYNAMIC = "'strict-dynamic'";
// referer values
exports.NO_REFERER = "'no-referer'";
exports.NONE_WHEN_DOWNGRADE = "'none-when-downgrade'";
exports.ORIGIN = "'origin'";
exports.ORIGIN_WHEN_CROSS_ORIGIN = "'origin-when-cross-origin'";
exports.UNSAFE_URL = "'unsafe-url'";
// sandbox values
exports.ALLOW_FORMS = "'allow-forms'";
exports.ALLOW_MODALS = "'allow-modals'";
exports.ALLOW_ORIENTATION_LOCK = "'allow-orientation-lock'";
exports.ALLOW_POINTER_LOCK = "'allow-pointer-lock'";
exports.ALLOW_POPUPS = "'allow-popups'";
exports.ALLOW_POPUPS_TO_ESACPE_SANDBOX = "'allow-popups-to-escape-sandbox'";
exports.ALLOW_PRESENTATION = "'allow-presentation'";
exports.ALLOW_SAME_ORIGIN = "'allow-same-origin'";
exports.ALLOW_SCRIPTS = "'allow-allow-scripts'";
exports.ALLOW_TOP_NAVIGATION = "'allow-top-navigation'";
exports.LIST_DIRECTIVES = {
    'child-src': 'child-src',
    'connect-src': 'connect-src',
    'default-src': 'default-src',
    'font-src': 'font-src',
    'form-action': 'form-action',
    'frame-ancestors': 'frame-ancestors',
    'frame-src': 'frame-src',
    'img-src': 'img-src',
    'manifest-src': 'manifest-src',
    'media-src': 'media-src',
    'object-src': 'object-src',
    'plugin-types': 'plugin-types',
    'prefetch-src': 'prefetch-src',
    'referrer': 'referrer',
    'sandbox': 'sandbox',
    'script-src': 'script-src',
    'style-src': 'style-src',
    'worker-src': 'worker-src'
};
exports.STRING_DIRECTIVES = {
    'base-uri': 'base-uri',
    'report-uri': 'report-uri'
};
exports.BOOLEAN_DIRECTIVES = {
    'block-all-mixed-content': 'block-all-mixed-content',
    'upgrade-insecure-requests': 'upgrade-insecure-requests',
};
exports.ALLOWED_DIRECTIVES = Object.assign({}, exports.LIST_DIRECTIVES, exports.STRING_DIRECTIVES, exports.BOOLEAN_DIRECTIVES);
