// express-csp-header/dist/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var csp_header_1 = require("csp-header");
exports.NONE = csp_header_1.NONE;
exports.SELF = csp_header_1.SELF;
exports.INLINE = csp_header_1.INLINE;
exports.EVAL = csp_header_1.EVAL;
exports.DATA = csp_header_1.DATA;
exports.BLOB = csp_header_1.BLOB;
exports.STRICT_DYNAMIC = csp_header_1.STRICT_DYNAMIC;
exports.NO_REFERER = csp_header_1.NO_REFERER;
exports.NONE_WHEN_DOWNGRADE = csp_header_1.NONE_WHEN_DOWNGRADE;
exports.ORIGIN = csp_header_1.ORIGIN;
exports.ORIGIN_WHEN_CROSS_ORIGIN = csp_header_1.ORIGIN_WHEN_CROSS_ORIGIN;
exports.UNSAFE_URL = csp_header_1.UNSAFE_URL;
exports.ALLOW_FORMS = csp_header_1.ALLOW_FORMS;
exports.ALLOW_MODALS = csp_header_1.ALLOW_MODALS;
exports.ALLOW_ORIENTATION_LOCK = csp_header_1.ALLOW_ORIENTATION_LOCK;
exports.ALLOW_POINTER_LOCK = csp_header_1.ALLOW_POINTER_LOCK;
exports.ALLOW_POPUPS = csp_header_1.ALLOW_POPUPS;
exports.ALLOW_POPUPS_TO_ESACPE_SANDBOX = csp_header_1.ALLOW_POPUPS_TO_ESACPE_SANDBOX;
exports.ALLOW_PRESENTATION = csp_header_1.ALLOW_PRESENTATION;
exports.ALLOW_SAME_ORIGIN = csp_header_1.ALLOW_SAME_ORIGIN;
exports.ALLOW_SCRIPTS = csp_header_1.ALLOW_SCRIPTS;
exports.ALLOW_TOP_NAVIGATION = csp_header_1.ALLOW_TOP_NAVIGATION;
exports.NONCE = '%nonce%';
exports.TLD = '%tld%';
